Userland/Libraries/LibWeb/Painting/ImagePaintable.cpp

```diff
@@ -65,7 +65,63 @@ void ImagePaintable::paint(PaintContext& context, PaintPhase phase) const
         } else if (auto bitmap = layout_box().image_provider().current_image_bitmap(image_rect.size().to_type<int>())) {
             ScopedCornerRadiusClip corner_clip { context, context.painter(), image_rect, normalized_border_radii_data(ShrinkRadiiForBorders::Yes) };
             auto scaling_mode = to_gfx_scaling_mode(computed_values().image_rendering(), bitmap->rect(), image_rect.to_type<int>());
-            context.painter().draw_scaled_bitmap(image_rect.to_type<int>(), *bitmap, bitmap->rect(), 1.f, scaling_mode);
+            auto object_fit = image_element.computed_css_values()->object_fit();
+            auto bitmap_rect = bitmap->rect();
+            auto bitmap_aspect_ratio = bitmap_rect.height() / bitmap_rect.width();
+            auto image_aspect_ratio = (float)image_rect.height().value() / image_rect.width().value();
+            auto image_int_rect = image_rect.to_type<int>();
+
+            auto scale_x = 0.0f;
+            auto scale_y = 0.0f;
+            Gfx::IntRect bitmap_intersect = bitmap_rect;
+
+            if (object_fit == CSS::ObjectFit::Fill) {
```

Comment:
Since every `ObjectFit` is handled here, using a switch would make it a compile error to miss one.

Reply:
I tried that in the beginning, got the message 'switch quantity not an integer' any idea why and what to do about it?

Reply:
Ah, it;s Optional<ObjectFit>! *facepalm

Reply:
Was making the same comment :)

`ScaleDown` case can even just fall-through to `None` to make it a bit clearer that we're making `ScaleDown` duplicate `None` for now.

```c++
switch (object_fit) {
case CSS::ObjectFit::Fill:
    ...

case CSS::ObjectFit::ScaleDown:
    // FIXME: Implement this.
    [[fallthrough]];

case CSS::ObjectFit::None:
    scale_x = 1;
    scale_y = 1;
    bitmap_intersect.set_size(image_int_rect.size());
    break;
}
```